server/src/main/java/org/candlepin/resource/validation/DTOValidator.java

```diff
@@ -0,0 +1,239 @@
+        Collection<?> collection1 = getter1.get();
+        Collection<?> collection2 = getter2.get();
+        Collection<?> collection3 = getter3.get();
+        if (isCollectionInvalid(collection1) ||
+            isCollectionInvalid(collection2) ||
+            isCollectionInvalid(collection3)) {
+            throw new IllegalArgumentException("collection contains null elements");
+        }
+    }
+
+    private boolean isCollectionInvalid(Collection<?> collection) {
+        return collection != null && collection.stream().anyMatch(Objects::isNull);
+    }
+
+
+
+    /**
+     * Accepts a variable amount of method references to getters that return Maps of elements, and
+     * validates that none of the elements (keys and values) contained in them are null. When at least one
+     * key or value of any of the Maps is null, an exception is thrown.
+     *
+     * If the map returned by the getter itself is null, no exception is thrown (null map is considered
+     * valid).
+     *
+     * Usage example:
+     * <pre>
+     * {@code
+     * validator.validateMapElementsNotNull(dto::getAttributes, dto::getProductAttributes);
+     * }
+     * </pre>
+     *
+     * @param getters a variable amount of method references to getters that return Maps of items
+     * @throws IllegalArgumentException when any of the maps returned from the specified getters contains at
+     * least one null key or value
+     */
+    @SafeVarargs
+    public final void validateMapElementsNotNull(Supplier<? extends Map<?, ?>> getter1,
+        Supplier<? extends Map<?, ?>> getter2, Supplier<? extends Map<?, ?>> getter3,
+        Supplier<? extends Map<?, ?>>... getters) {
+
+        validateMapElementsNotNull(getter1, getter2, getter3);
+
+        for (Supplier<? extends Map<?, ?>> getter : getters) {
+            Map<?, ?> map = getter.get();
+            if (isMapInvalid(map)) {
+                throw new IllegalArgumentException("map contains null elements");
+            }
+        }
+    }
+
+    /*
+     * ======================================================================================================
+     * The following methods are overloaded versions of validateMapElementsNotNull for one, two and
+     * three getters respectively, as a performance optimization to avoid the implicit creation of an array
+     * object that happens during the vararg method call. There are currently no DTOs that contain more than
+     * three maps that require null element validation, so this optimisation would cover 100% of cases
+     * for now, while allowing us to validate more than 3 maps at a time in the future.
+     * ======================================================================================================
+     */
+
+    /**
+     * Works the same as {@link #validateMapElementsNotNull(Supplier, Supplier, Supplier, Supplier...)}
+     */
+    public void validateMapElementsNotNull(Supplier<? extends Map<?, ?>> getter) {
```

Comment:
Same deal here: use this version as the base, and have the extended versions just call into it with each arg.

Reply:
Done